### PR TITLE
feat: add section component

### PR DESCRIPTION
**Pull request by nmerget**

### CODE REVIEW

File: packages/components/src/components/page/page.scss [hunk 1]
```diff
@@ -50,11 +40,11 @@
...
 			flex: 1 1 auto;
 			width: 100%;
+		}
 
-			& > * {
-				overflow: auto;
-				height: 100%;
-			}
+		main > :first-child {
+			overflow: auto;
```
Comment (mfranzke) [inline packages/components/src/components/page/page.scss:46]: what for is this `overflow: hidden` necessary?

Comment (nmerget) [reply packages/components/src/components/page/page.scss]: you are right, with sections we don't need it anymore

File: packages/components/src/components/section/section.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
+@use "@db-ui/foundations/build/scss/variables" as *;
+@use "@db-ui/foundations/build/scss/variables.global" as *;
+@use "@db-ui/foundations/build/scss/helpers/functions" as *;
+
+.db-section {
+	width: 100%;
+
+	padding-left: $db-spacing-fixed-md;
+	padding-right: $db-spacing-fixed-md;
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:9]:
```suggestion
	padding-inline: $db-spacing-fixed-md;
```

File: packages/components/src/components/section/section.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	@media screen and (min-width: $db-screens-m) {
+		padding-left: $db-spacing-fixed-lg;
+		padding-right: $db-spacing-fixed-lg;
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:13]:
```suggestion
		padding-inline: $db-spacing-fixed-lg;
```

File: packages/components/src/components/section/section.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	&[data-variant="full"] {
+		padding-left: 0;
+		padding-right: 0;
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:18]:
```suggestion
		padding-inline: 0;
```

File: packages/components/src/components/section/section.scss [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	&[data-size="none"] {
+		padding-top: 0;
+		padding-bottom: 0;
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:23]:
```suggestion
		padding-block: 0;
```

File: packages/components/src/components/section/section.scss [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	&[data-size="small"] {
+		padding-top: $db-spacing-responsive-sm;
+		padding-bottom: $db-spacing-responsive-sm;
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:28]:
```suggestion
		padding-block: $db-spacing-responsive-sm;
```

Comment (mfranzke) [inline packages/components/src/components/section/section.scss:28]:
```suggestion
		padding-block: $db-spacing-responsive-sm;
```

File: packages/components/src/components/section/section.scss [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	&[data-size="medium"] {
+		padding-top: $db-spacing-responsive-md;
+		padding-bottom: $db-spacing-responsive-md;
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:33]:
```suggestion
		padding-block: $db-spacing-responsive-md;
```

File: packages/components/src/components/section/section.scss [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+	}
+
+	&[data-size="large"] {
+		padding-top: $db-spacing-responsive-lg;
+		padding-bottom: $db-spacing-responsive-lg;
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:38]:
```suggestion
		padding-block: $db-spacing-responsive-lg;
```

File: packages/components/src/components/section/section.scss [hunk 9]
```diff
@@ -0,0 +1,57 @@
+@use "@db-ui/foundations/build/scss/variables" as *;
+@use "@db-ui/foundations/build/scss/variables.global" as *;
+@use "@db-ui/foundations/build/scss/helpers/functions" as *;
+
+.db-section {
+	width: 100%;
+
```
Comment (mfranzke) [inline packages/components/src/components/section/section.scss:7]:
```suggestion
```

this element is of type `display: block`, so it's 100% width anyhow, or am I misunderstanding something here?

File: packages/components/src/components/page/page.scss [hunk 10]
```diff
@@ -47,14 +37,9 @@
 		main {
 			display: flex;
 			flex-direction: column;
-			overflow: hidden;
+			overflow: auto;
```
Comment (mfranzke) [inline packages/components/src/components/page/page.scss:40]: I might have asked this question previously – why do we need any `overflow: *` in the main landmarks of our page at all (and reset it again with this declaration)?

Comment (nmerget) [reply packages/components/src/components/page/page.scss:40]: Let's postpone this. We had a discussion with UX about it. 
We will add variants to the header an footer like "fixed". And also for db-page to configure what we want.

For now I just copy paste this from elements. Let's change it when we add the v3 stylings for header and footer

**Comment by mfranzke** [comment]
are you leaving out the other showcases intentionally? Could you please write a "follow up" issue for them than?

File: packages/components/src/components/section/section.scss (hunk 9 above)
Comment (mfranzke) [reply packages/components/src/components/section/section.scss:7]: @nmerget how is this resolved?

**Review by mfranzke — COMMENTED** [review]
are you leaving out the other showcases intentionally? Could you please write a "follow up" issue for them than?

**Comment by nmerget** [comment]
> are you leaving out the other showcases intentionally? Could you please write a "follow up" issue for them than?

Yes I we should do it after unifying the showcases:
https://github.com/db-ui/mono/issues/375

File: packages/components/src/components/page/page.scss (hunk 10 above)
Comment (mfranzke) [reply packages/components/src/components/page/page.scss:41]: https://github.com/db-ui/mono/issues/376

**Review by mfranzke — APPROVED** [review]

**Pull request merged**